great_expectations/self_check/util.py

```diff
@@ -466,6 +509,33 @@ def get_dataset(
         return SqlAlchemyDataset(
             custom_sql=custom_sql, engine=engine, profiler=profiler, caching=caching
         )
+    elif dataset_type == "bigquery":
+        if not create_engine:
+            return None
+        engine = _create_bigquery_engine()
+
+        schema = None
+        if schemas and dataset_type in schemas:
+            schema = schemas[dataset_type]
+
+        # BigQuery does not allow for column names to have spaces
+        if schema:
+            schema = {k.replace(" ", "_"): v for k, v in schema.items()}
```

Comment:
The `if schema` can only ever run if the above conditional is `True` right? Could we consolidate to:
```suggestion
        schema = None
        if schemas and dataset_type in schemas:
            schema = schemas[dataset_type]
            # BigQuery does not allow for column names to have spaces
            if schema:
                schema = {k.replace(" ", "_"): v for k, v in schema.items()}
```
And if `schema = schemas[dataset_type]` guarantees us a non-`NoneType`, we could get rid of the second `if schema`?

```suggestion
        schema = None
        if schemas and dataset_type in schemas:
            schema = schemas[dataset_type]
            # BigQuery does not allow for column names to have spaces
            schema = {k.replace(" ", "_"): v for k, v in schema.items()}
```